subsys/esb/esb.c

```diff
@@ -885,11 +921,35 @@ static void start_tx_transaction(void)
 	nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_ADDRESS);
 	nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_PAYLOAD);
 	nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_DISABLED);
+	nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_END);
 
-	esb_ppi_for_txrx_set(false, ack);
-	esb_fem_for_tx_set(ack);
+	/* Trigger different radio event if radio is disabled or idle */
+	if (is_tx_idle) {
+		nrf_radio_task_trigger(NRF_RADIO, NRF_RADIO_TASK_START);
+	} else {
+		esb_ppi_for_txrx_set(false, ack);
+		esb_fem_for_tx_set(ack);
 
-	radio_start();
+		radio_start();
+	}
+}
+
+static void on_radio_end_tx_noack(void)
+{
+	/* Timer compare is cleared by PPI - we still need to disable Interrupt flag */
+	nrf_timer_int_disable(esb_timer.p_reg,  nrf_timer_compare_int_get(NRF_TIMER_CC_CHANNEL1));
+	esb_ppi_for_wait_for_rx_clear();
+
+	interrupt_flags |= INT_TX_SUCCESS_MSK;
+	tx_fifo_remove_last();
+
+	if (tx_fifo.count == 0) {
+		esb_state = ESB_STATE_PTX_TXIDLE;
+		NVIC_SetPendingIRQ(ESB_EVT_IRQ);
```

Comment:
Can't the RADIO be disabled if there are no packets in the FIFO? With fast ramp-up configured, the transmission will be relatively fast when application decides to send a new packet.

Reply:
This is a good remark.
Although, my application requires under 100µs delay between fifo data push and reception.
Even with fast ramp-up activated, this is not achievable from radio_off state.
Thus, even if there is break in the fifo stream, i do not wish to ramp-up.

This is why i called the parameter "never_disable_tx"